src/fspec-archive.c

```diff
@@ -80,9 +98,15 @@ int main(int argc, char **argv)
             } else if (strncmp(line, "gid=", 4) == 0) {
                 archive_entry_set_gid(entry, strtol(line + 4, NULL, 10));
             } else if (strncmp(line, "mode=", 5) == 0) {
-                archive_entry_set_perm(entry, strtol(line + 5, NULL, 8));
+                set_default_mode = 0;
+                mode = strtol(line + 5, NULL, 8);
             } else if (strncmp(line, "type=", 5) == 0) {
-                archive_entry_set_filetype(entry, filetype(line + 5));
+                const char *t = line + 5;
+
+                archive_entry_set_filetype(entry, filetype(t));
+                if (set_default_mode) {
+                    mode = defaultmode(t);
+                }
```

Comment:
I think this should be moved to after the attribute parsing loop, so that `mode` and `type` could appear in either order.

Reply:
Err... never mind, how it is currently is fine.